tianshou/data/batch.py

```diff
@@ -652,8 +643,8 @@ def __len__(self) -> int:
         for v in self.__dict__.values():
             if isinstance(v, Batch) and v.is_empty(recurse=True):
                 continue
-            elif hasattr(v, '__len__') and (not isinstance(
-                    v, (np.ndarray, torch.Tensor)) or v.ndim > 0):
+            elif hasattr(v, '__len__') and (
```

Comment:
Are you sure it is equivalent / better ?

Reply:
Since the type of `v` is (Batch, np.ndarray, torch.Tensor) (same as _parse_value), if v is not (np.ndarray and torch.Tensor), we can assume that v is a Batch.
By the way, if I use the original code-snippets, mypy will raise an error: 
```
tianshou/data/batch.py: note: In member "__len__" of class "Batch":
tianshou/data/batch.py:646: error: Left operand of 'or' is always false  [unreachable]
                elif hasattr(v, '__len__') and (not isinstance(
```